ctoken-monitor/src/autotask.js

```diff
@@ -0,0 +1,330 @@
+  } else {
+    const underlyingTokenContract = new ethers.Contract(
+      underlyingTokenAddress,
+      TOKEN_ABI,
+      provider,
+    );
+    decimals = await underlyingTokenContract.decimals();
+    symbol = await underlyingTokenContract.symbol();
+  }
+  return { decimals, symbol };
+}
+
+function formatAmountString(amount, decimals) {
+  const amountBN = ethers.BigNumber.from(amount);
+  const divisorBN = ethers.BigNumber.from(10).pow(decimals);
+
+  // the ethers BigNumber implementation will discard the decimal
+  // portion of the value when we perform the division
+  let resultString = amountBN.toString();
+  if (resultString.length <= decimals) {
+    resultString = `0.${'0'.repeat(decimals - resultString.length)}${resultString[0]}`;
+  } else {
+    resultString = amountBN.div(divisorBN).toString();
+  }
+
+  // format the number to have comma separators for powers of 1000
+  const internationalNumberFormat = new Intl.NumberFormat('en-US');
+  return internationalNumberFormat.format(resultString);
+}
+
+function createDiscordMessage(eventName, metadata, decimals, symbol, description) {
+  const eventObject = eventMapping[eventName];
+  if (eventObject !== undefined) {
+    const amount = metadata[eventObject.amountKey];
+    const amountString = formatAmountString(amount, decimals);
+    const byAddress = metadata[eventObject.byKey];
+    const { action } = eventObject;
+    let message = `**${amountString} ${symbol}** ${action}`;
+
+    if (action === 'liquidate') {
+      const fromAddress = metadata[eventObject.fromKey];
+      message += ` from ${fromAddress.slice(0, 6)} by ${byAddress.slice(0, 6)}`;
+    } else if (action === 'repayBorrow') {
+      const forAddress = metadata[eventObject.forKey];
+      message += ` by ${byAddress.slice(0, 6)}`;
+      if (forAddress !== byAddress) {
+        message += ` for ${forAddress.slice(0, 6)}`;
+      }
+    } else {
+      message += ` by ${byAddress.slice(0, 6)}`;
+    }
+
+    const emoji = description.slice(0, description.indexOf('-') - 1);
+
+    return `${emoji} ${message}`;
+  }
+  return undefined;
+}
+
+async function post(url, method, headers, data) {
+  return axios({
+    url, method, headers, data,
+  });
+}
+
+async function postToDiscord(url, message) {
+  const method = 'post';
+  const headers = {
+    'Content-Type': 'application/json',
+  };
+  const data = JSON.stringify({ content: message });
+
+  let response;
+  try {
+    // perform the POST request
+    response = await post(url, method, headers, data);
+  } catch (error) {
+    // is this a "too many requests" error (HTTP status 429)
+    if (error.response && error.response.status === 429) {
+      // the request was made and a response was received
+      // try again after waiting 5 seconds
+      // eslint-disable-next-line no-promise-executor-return
+      const promise = new Promise((resolve) => setTimeout(resolve, 5000));
+      await promise;
+      response = await post(url, method, headers, data);
```

Comment:
```suggestion
      try {
        response = await post(url, method, headers, data);
      } catch (error2) {
        // Handle error
      }
```
What if this errors out?

Reply:
Actually, your way is good. Ref: https://eslint.org/docs/rules/no-useless-catch